src/frontend/frontend/views/publisher_views.py

```diff
@@ -69,10 +69,44 @@ class ProductTypeView(BaseView):
     @classmethod
     def get_extra_context(cls, object_id: int | str) -> dict[str, Any]:
         dpl = DataPersistenceLayer()
+        templates = dpl.get_objects(Template)
+        template_files = []
+        try:
+            from frontend.core_api import CoreApi
+            core_api = CoreApi()
+            api_result = core_api.api_get("/config/templates", params={"list": True})
+            validation_map = {}
+            if api_result and "items" in api_result:
+                for item in api_result["items"]:
+                    template_id = item.get("id")
+                    validation_status = item.get("validation_status", {})
+                    is_valid = validation_status.get("is_valid", None)
+                    if is_valid is True:
+                        status = "valid"
+                    elif is_valid is False:
+                        status = "invalid"
+                    else:
+                        status = "unknown"
+                    validation_map[template_id] = status
+            for t in templates:
+                status = validation_map.get(t.id, "unknown")
+                template_files.append({
+                    "id": t.id,
+                    "name": t.id,
+                    "validation_status": status
+                })
+        except Exception as e:
+            logger.warning(f"Failed to get template validation status from API: {e}")
+            for t in templates:
+                template_files.append({
+                    "id": t.id,
+                    "name": t.id,
+                    "validation_status": "unknown"
+                })
```

Comment:
If included in the Template response this whole block could be replaced by an adaption of the old code:

            "template_files": [{"id": t.id, "name": t.id, "validation_state": t.validation_state} for t in dpl.get_objects(Template)],